ui/lib/CreateStream.tsx

```diff
@@ -71,62 +189,259 @@ const CreateStream = (props: CreateStreamProps) => {
     functionName: "approve",
     args: [sm.address, selectedTime * props.amountPerSecond],
   });
-  const { write: approveStream } = useContractWrite(approvalConfig);
+
+  const {
+    data,
+    isLoading,
+    // isSuccess: Make sure transaction has been approved by user (used to get tx hash)
+    isSuccess,
+    isError,
+    write: approveStream,
+  } = useContractWrite(approvalConfig);
+
+  // Then make sure transaction has been processed once it has been approved by user
+  const [txHash, setTxHash] = useState<string | undefined>(undefined);
+  useEffect(() => {
+    if (isSuccess && data?.hash) {
+      setTxHash(data?.hash);
+    }
+  }, [isSuccess, data?.hash]);
+
+  const {
+    error: txError,
+    isSuccess: txSuccess,
+    isLoading: txLoading,
+  } = useWaitForTransaction({
+    hash: txHash as `0x${string}`,
+  });
+
   const createStream = () => {
-    // NOTE: This function should move away from this component
-    sm.create(props.tokenAddress, props.amountPerSecond, props.reasonCode).then(
-      props.registerStream,
-    );
+    props.handleTransactionStatus(true, false, null);
+
+    props
+      .renderReasonCode()
+      .then((reasonString) => {
+        return sm
```

Comment:
nothing to return really

```suggestion
        sm
```